spec/helpers/item_seasons_helper_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ItemSeasonsHelper, type: :helper do
+  describe '#rotation' do
```

Comment:
**question:** What happens when we give an unexpected number (let's say 52). What would happen? Should we add a test case for this?

Reply:
Hum, good question!

```
<div class="button-holder p-top-l p-bottom-l p-sides-s">
    <% ItemSeason::MONTHS.each do |month_index| %>
      <%= link_to (formatted_month_short(month_index)).capitalize, item_season_path(month_index),
          class: "link_action month_link button button--yellow script-lg m-top-s m-bottom-s #{rotation(month_index)}" %>
    <%  end %>
  </div>
```

This is what we have right now. If we got a 52-month-season in the rotation, it would rotate to 0. However, we don't want an extra month to show at all. So I think there is a bigger problem here.

I made some tests and I think a way of dealing with it might be only running if `month_index <= 12`. This way we make sure we are only ever showing 12 months.

```
  <div class="button-holder p-top-l p-bottom-l p-sides-s">
    <% ItemSeason::MONTHS.each do |month_index| %>
      <% if month_index <= 12 %>
        <%= link_to (formatted_month_short(month_index)), item_season_path(month_index),
              class: "link_action month_link button button--yellow script-lg m-top-s m-bottom-s #{rotation(month_index)}" %>
      <%  end %>
    <% end %>
  </div>
```

Then we could maybe add a test case to views or features to check if we only have 12 months.

WDYT?

Reply:
You went way deeper than what I expected :)

### About my question per se

I was talking  purely from a code perspective. The function knows nothing of your restrictions. So from a functional (and testing) perspective we should make sure that when fed with an unexpected number, it returns something you expect (in this case, `0` rotation). 

So my objective was just making you a test case that covered this.

### About where the question branched to

There's no need to do a check, you control the constant. If there are more numbers there, then that should be dealt directly in the constant. You just need to add guards and all when you do not control the data (i.e: it comes from user input).

Reply:
I added the test and removed the check 🙂 

But I have a question: even if we don't want this to run at all, do we still need to do the test? Because, if it goes through we expect it to return zero, but in the end, it shouldn't get to that point. So I'm confused about why should we test for something that shouldn't happen.

Reply:
Because even though it shouldn't, the code allows for it. Assuming you work in a bigger team, anyone could misuse the method. There's nothing there explicitly saying you expect numbers from 1 to 12.

That said, i think my point was misguided, there are some intervals between 1 and 12 that do not have a specific rotation, so this case is already covered :)

Sorry for that

Reply:
But to answering the "if we don't want this to run at all" comment: if we don't want it to run outside of expected parameters, the best thing to do would be to raise an error (i.e: `raise 'index must be between 1 and 12`]

Reply:
Added the error to the helper and to the test. Though I'm not sure if the test I wrote is quite right. I wanted to test for any number bigger than 12, so I used rand, but I had to use a max value.

Reply:
You should test for numbers smaller than 1 and bigger than 12. That said, you do not need to get a random number, just the first one.